### DISCUSSION TITLE
Instruct Mode through API extension?

### DISCUSSION THREAD

**Discussion by elindiotaino** [oobabooga/text-generation-webui#1137] [Q&A]
How can I use the instruct mode when making requests through the API extension?

This is my python code for the API call:

```py
import requests 
import json  
data = {     
    "prompt": "Below is an instruction that describes a task. Write a response that appropriately completes the request.\n### Human:\nWhat is the capital of USA?\n### Assistant:",     
    "max_new_tokens": 100,
    "do_sample": True,
    "temperature": 0.85,
    'top_p': 1,
    'typical_p': 1,
    'repetition_penalty': 1.01,
    'encoder_repetition_penalty': 1,
    'top_k': 50,
    'min_length': 0,
    'no_repeat_ngram_size': 0,
    'num_beams': 1,
    'penalty_alpha': 0,
    'length_penalty': 1,
    'early_stopping': False,
    'seed': -1,
    'add_bos_token': True,
    'truncation_length': 2048,
    'ban_eos_token': False,
}  
headers = {     
"Content-Type": "application/json" 
}  
response = requests.post("http://127.0.0.1:5000/api/v1/generate", data=json.dumps(data), headers=headers)  
if response.status_code == 200:     
    results = json.loads(response.content)["results"]     
    for result in results:         
        print(result["text"]) 
else:     
    print(f"Request failed with status code {response.status_code}.") 
```

**Comment by cibernicola** [comment]
Provide more information, tested your example and it works ("technically") fine: 
![image](https://user-images.githubusercontent.com/4579387/231860905-0e81893e-dd09-42b3-ba29-e362dd5fd116.png)

**Comment by umLu** [comment]
It seemed to me more beneficial for the API extension to actually work as if it were in "instruct" mode, so I made a modification to it and proposed a pull request. See if it makes sense.

#1365

**Reply by danpizzy** [reply]
how are yo using the api it keeps giving me this error ive tried many commits import requests 
Traceback (most recent call last):
  File "d:\ai scraper gpt\hugging.py", line 54, in <module>
    }).json()
  File "C:\Users\KAMII\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\requests\models.py", line 975, in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
